source/components/GlobalNavigation/components/GlobalNavigationNotifications/NotificationsDataProvider.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+
+import Api from '../../utils/api';
+import { NotificationsProvider } from '../../utils/NotificationContext';
+import Notification from '../../models/Notification';
+
+/* eslint-disable react/no-unused-state */
+class NotificationsDataProvider extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.api = Api.build(props.serviceUrl);
+
+        this.state = {
+            unreadCount: 0,
+            notifications: [],
+            isLoading: false,
+            firstPageLoaded: false,
+            nextPage: null,
+            loadFirstPage: this.loadFirstPage.bind(this),
+            loadNextPage: this.loadNextPage.bind(this),
+            markAsRead: this.markAsRead.bind(this),
+            markAllAsRead: this.markAllAsRead.bind(this),
+            goToDestination: this.goToDestination.bind(this),
+        };
+    }
+
+    componentDidMount() {
+        this.loadUnreadNotificationCount();
+    }
+
+    getNextPageLink(data) {
+        return get(data, '_links.next') || null;
+    }
+
+    getUnreadCount(notifications) {
+        return notifications.filter(notification => notification.isUnread).length;
+    }
+
+    goToDestination(latestEventUri) {
+        // TODO check if it's a local url and use transitionTo or delegate to app
+        window.location.href = latestEventUri;
+    }
+
+    loadFirstPage() {
+        const { nextPage, firstPageLoaded } = this.state;
+
+
+        if (nextPage !== null || firstPageLoaded) {
+            return;
+        }
+
+        this.loadPage('/notifications?contentType=discussion-upvote&contentType=discussion-post&contentType=announcement-target');
+    }
+
+    loadNextPage() {
+        const { nextPage } = this.state;
+
+        if (nextPage === null) {
+            return;
+        }
+
+        this.loadPage(nextPage);
+    }
+
+    loadPage(pageLink) {
+        const { isLoading } = this.state;
+
+        if (isLoading) {
+            return;
+        }
+
+        this.setState({ isLoading: true, firstPageLoaded: true });
+
+        this.api
+            .loadPage(pageLink)
+            .then((response) => {
+                const mappedNotifications = this.mapNotifications(response.notifications);
+
+                this.setState({
+                    notifications: mappedNotifications,
+                    unreadCount: this.getUnreadCount(mappedNotifications),
+                    nextPage: this.getNextPageLink(response),
+                    isLoading: false,
+                });
+            })
+            .catch((error) => {
+                this.setState({ isLoading: false });
+
+                console.error('Couldn\'t load notifications', error);
```

Comment:
At some point we should probably add some logging library to react-commons which plugs into event-logger. This ticket is already a huge amount of work though, so no need to add to it this time :)

Reply:
agree. i thought about it now, but there's still plenty of work to do regarding GN, so i decided to leave it